lambda_src/lambda_function.py

```diff
@@ -33,19 +38,22 @@ def async_flow_init(event: Any, context: Any) -> Dict[Text, Any]:
     LOG.debug('Found a destination header and hence using async_flow_init().')
 
     headers = event['headers']
+    req_body = loads(event['body'])
     batch_id = headers[BATCH_ID_HEADER]
     destination = headers[DESTINATION_URI_HEADER]
-    headers.pop(DESTINATION_URI_HEADER)
     headers['write-uri'] = destination
     lambda_name = context.function_name
     LOG.debug(f'async_flow_init() received destination: {destination}.')
 
     destination_driver = import_module(
         f'geff.drivers.destination_{urlparse(destination).scheme}'
     )
-    # Ignoring style due to dynamic import
-    destination_driver.initialize(destination, batch_id)  # type: ignore
+    # Ignoring style due to dynamic imports
+    for rn, *args in req_body['data']:
+        parameterized_destination = format(destination, args)
+        destination_driver.initialize(parameterized_destination, batch_id)  # type: ignore
```

Comment:
I think you realizes this changes the contract between the main GEFF lambda processor and the write drivers, calling initialize once per row instead of once per EF call. I think we talked about this so I imagine you have a reason here, but I'm not sure what it is, and it seems it could add a lot of overhead at scale.

might it make sense to start out by requiring that the initialize call not be parametrized by the row args, only by the keywords we're adding, and avoid the change to the driver interface?